### DISCUSSION TITLE
VScode extension not working, cli is

### DISCUSSION THREAD

**Discussion by cdietrich** [eclipse-langium/langium#967] [Q&A] [answered]
I created a langium project and wrote the grammar. the cli is working, but vscode does not show any problems. any hints where to start debugging?

https://github.com/cdietrich/core-dsl-langium-vscode-not-working

to me it looks in in vscode the extension dislikes my terminal(fragments)
unfortunately converting them to regexes is very tedious

2nd question: is there an easy way to test lexing?

**Comment by msujew** [comment, marked as answer]
Compared to Xtext, Langium has a slightly different implementation for the `!` terminal element. See also https://github.com/langium/langium/issues/681. In your case, it leads to an infinite recursion while lexing. You just need to change these terminals:
```
terminal CHARCONST:	('u'|'U'|'L')? "'" ( '\\' . /* 'b'|'t'|'n'|'f'|'r'|'u'|'"'|"'"|'\\' */ | !('\\'|"'") )* "'";
terminal ENCSTRINGCONST	:	('u8'|'u'|'U'|'L') '"' ( '\\' . /* 'b'|'t'|'n'|'f'|'r'|'u'|'"'|"'"|'\\' */ | !('\\'|'"') )* '"';
```
to:
```
terminal CHARCONST:	('u'|'U'|'L')? "'" ( '\\' . /* 'b'|'t'|'n'|'f'|'r'|'u'|'"'|"'"|'\\' */ | /[^\\']/ )* "'";
terminal ENCSTRINGCONST	:	('u8'|'u'|'U'|'L') '"' ( '\\' . /* 'b'|'t'|'n'|'f'|'r'|'u'|'"'|"'"|'\\' */ | /[^\\"]/ )* '"';
```

2. We directly expose the `Lexer` from the `services.parser.Lexer` DI container. There's no builtin testing facility, but the instance you get back from the DI container should be enough.

**Reply by cdietrich** [reply]
thx this works. still had to resort some stuff as the langium lexer also seems to have differently regarding order/greedyness to the antlr one https://github.com/cdietrich/CoreDsl-Langium/commit/2570dd7625dcd01751008ac956b1592c4f83a250#diff-6a42cc78a2dc481c3263b6194dc80001272fb3debf866c18df2ad6557a70cbdf
